File: app/src/main/java/com/amplifyframework/datastore/sample/RemotePresentation.java
package com.amplifyframework.datastore.sample;

import com.amplifyframework.datastore.appsync.ModelWithMetadata;
import com.amplifyframework.datastore.generated.model.Post;

import io.reactivex.Observable;
import io.reactivex.Single;

interface RemotePresentation {
    interface View {
        void displayRemoteLogLine(LogLine logLine);
        void clearRemoteLineItems();
    }

    interface Presenter {
        void createRemotePost();
        void updateRemotePosts();
        void deleteRemotePosts();
        void listRemotePosts();
        void clearRemoteLogs();
    }

    interface ApiInteractor {
        Single<ModelWithMetadata<Post>> createRandom();
        Single<ModelWithMetadata<Post>> update(ModelWithMetadata<Post> post);
        Single<ModelWithMetadata<Post>> delete(ModelWithMetadata<Post> post);
        Observable<ModelWithMetadata<Post>> list();
    }
}

File: app/src/main/java/com/amplifyframework/datastore/sample/RemotePresentation.kt
package com.amplifyframework.datastore.sample

import com.amplifyframework.datastore.appsync.ModelWithMetadata
import com.amplifyframework.datastore.generated.model.Post

import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.core.Single

internal interface RemotePresentation {
    interface View {
        fun displayRemoteLogLine(logLine: LogLine)
        fun clearRemoteLineItems()
    }

    interface Presenter {
        fun createRemotePost()
        fun updateRemotePosts()
        fun deleteRemotePosts()
        fun listRemotePosts()
        fun clearRemoteLogs()
    }

    interface ApiInteractor {
        fun createRandom(): Single<ModelWithMetadata<Post>>
        fun update(post: ModelWithMetadata<Post>): Single<ModelWithMetadata<Post>>
        fun delete(post: ModelWithMetadata<Post>): Single<ModelWithMetadata<Post>>
        fun list(): Observable<ModelWithMetadata<Post>>
    }
}
